### PR TITLE
Add World Switcher Plugin

### PR DESCRIPTION
**Pull request by psikoi**
Adds the full UI implementation for a world switcher plugin.

This includes:

- World list, with fancy new country icons, players and activity indicators.
- World sorting, by world, players or activity
- Adding world to favorites
- Removing world from favorites
- Double clicking to change world
- PVP / Tournament highlighting
- Current world highlighting

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/worldswitcher/WorldSwitcherPlugin.java [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+		}
+
+		navButton = NavigationButton.builder()
+			.tooltip("World Switcher")
+			.icon(icon)
+			.priority(3)
+			.panel(panel)
+			.build();
+
+		clientToolbar.addNavigation(navButton);
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/worldswitcher/WorldSwitcherPlugin.java]: This entire sidebar should be disable-able in plugin config as if this plugin is also going to have the world hop shortcuts and hop-to, it should not require to have the sidebar present when there is already better version in-game.

**Comment by deathbeam** [comment]
For the arrow-down and arrow-faded, use ImageUtil. All resources you need here is arrow pointing up and then just transform it to other variants.

File: runelite-client/src/main/java/net/runelite/client/plugins/worldswitcher/WorldSwitcherPlugin.java (hunk 1 above)
Comment (psikoi) [reply runelite-client/src/main/java/net/runelite/client/plugins/worldswitcher/WorldSwitcherPlugin.java]: What's the solution here, then?

**Comment by psikoi** [comment]
Didn't know ImageUtil existed tbh

File: runelite-client/src/main/java/net/runelite/client/plugins/worldswitcher/WorldSwitcherPlugin.java (hunk 1 above)
Comment (deathbeam) [reply runelite-client/src/main/java/net/runelite/client/plugins/worldswitcher/WorldSwitcherPlugin.java]: Config option for disabling the sidebar I guess

Comment (psikoi) [reply runelite-client/src/main/java/net/runelite/client/plugins/worldswitcher/WorldSwitcherPlugin.java]: Which is?

**Comment by Nightfirecat** [comment]
It's new. :man_shrugging:

**Comment by Gamer1120** [comment]
It might be useful if this takes the login limit into account, such that it will show a warning that hopping would exceed the login limit and how much time it takes before you can hop again. Saves the trouble of re-typing your password. This could also be a separate issue.

**Comment by psikoi** [comment]
@Gamer1120 is this timeout an exact amount of time?

**Comment by Gamer1120** [comment]
@psikoi I'm not 100% sure which time it is exactly, but I think it's that you can login X times per X seconds. I'll try to Google if someone already figured that out.

**Comment by Gamer1120** [comment]
https://www.reddit.com/r/2007scape/comments/533qdl/world_hop_limit/ is not 100% sure, but this could be used as a basis. I'll try to test it.

**Comment by Adam-** [comment]
Merge as part of #4761

**Pull request closed without merging**